src/db.rs

```diff
@@ -718,6 +723,13 @@ impl DbInner {
 					}
 					log::debug!(target: "parity-db", "Log queue size: {} bytes", *queue);
 				}
+
+				let max_logs = if self.options.sync_data { MAX_LOG_FILES } else { KEEP_LOGS };
+				let dirty_logs = self.log.num_dirty_logs();
+				if dirty_logs > max_logs {
```

Comment:
```suggestion
				while self.log.num_dirty_logs() > max_logs {
```
in the sense that cleanup can signal if incomplete clean. But could also be better not to slow down in this case.